src/components/TodoAppFooter.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useContext } from 'react';
+import { DispatchContext, StateContext } from '../context/ContextReducer';
+import cn from 'classnames';
+
+export const TodoAppFooter: React.FC = () => {
+  const { select, totalLength } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const itemLeft = totalLength.filter(todo => !todo.completed);
+  const clear = totalLength.some(todo => todo.completed);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${itemLeft.length} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          onClick={() => dispatch({ type: 'setSelect', value: 'All' })}
+          href="#/"
+          className={cn('filter__link', { selected: select === 'All' })}
```

Comment:
move these values to some enum 
```suggestion
          className={cn('filter__link', { selected: select === FilterStatus.All })}
```